### PR TITLE
Python code: Remove superfluous parens

### PR DESCRIPTION
**Pull request by bje-**
## What does this PR do?

Silences Pylint warnings about superfluous parentheses.

### CODE REVIEW

File: autotest/ogr/ogr_osm.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ def ogr_osm_5():
         feat = None
         ds.ReleaseResultSet(sql_lyr)
 
-        if not (test[2] ^ is_none):
+        if not test[2] ^ is_none:
```
Comment (rouault) [inline autotest/ogr/ogr_osm.py]: This change assumes everybody knows operator precedence. I don't... Less risky to have the extra parenthesis

File: autotest/osr/osr_esri.py [hunk 2]
```diff
@@ -1214,7 +1214,7 @@ def osr_esri_25():
     expected_proj4_string = ('+a=6378137 +b=6378137 +proj=merc +lat_ts=0'
                              ' +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +no_defs')
     proj4_string = srs.ExportToProj4()
-    if not (expected_proj4_string.split(' ').sort() == proj4_string.split(' ').sort()):
+    if not expected_proj4_string.split(' ').sort() == proj4_string.split(' ').sort():
```
Comment (rouault) [inline autotest/osr/osr_esri.py]: Same here. Parenthesis help removing any ambiguity

File: gdal/swig/python/samples/dump_jp2.py [hunk 3]
```diff
@@ -136,7 +136,7 @@ def extract_all_xml_boxes(filename, prefix):
                     end_gmljp2_link = end_gmljp2_link_double_quote
                 if end_gmljp2_link >= 0:
                     referenced_box = mdd_item[new_pos + len('gmljp2://xml/'):end_gmljp2_link]
-                    if not (('xml:' + referenced_box) in mdd_list):
+                    if not 'xml:' + referenced_box in mdd_list:
```
Comment (rouault) [inline gdal/swig/python/samples/dump_jp2.py]: Same here

Comment (bje-) [reply gdal/swig/python/samples/dump_jp2.py]: I'll meet you halfway on this one. :-)

**Pull request merged**